torch/autograd/function.py

```diff
@@ -98,6 +98,9 @@ def __init__(cls, name, bases, attrs):
 
         # old-style functions
         if not has_static_forward:
+            warnings.warn('Legacy autograd function with non-static forward method is deprecated. '
```

Comment:
I'm wondering if test_autograd.py will warning spew. At some point we should convert the legacy functions there that are not explicitly testing legacy behavior to non-legacy ones (probably doesn't have to happen in this PR)

Reply:
That's a good idea, I opened another PR for it: https://github.com/pytorch/pytorch/pull/22925.